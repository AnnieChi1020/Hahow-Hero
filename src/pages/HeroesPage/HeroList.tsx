import React from 'react';
import styled from 'styled-components';
import HeroCard from './HeroCard';
import { Hero } from '../../redux/heroSlice';

const HeroList: React.FC<{ heroList: Hero[] }> = (props) => {
    const { heroList } = props;

    return (
        <Container>
            {heroList.map((h) => (
                <HeroCard key={h.id} hero={h} />
            ))}
        </Container>
    );
};

export default HeroList;

const Container = styled.div`
    display: flex;
    gap: 24px;
`;
