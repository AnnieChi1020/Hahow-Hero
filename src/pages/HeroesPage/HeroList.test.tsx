import React from 'react';
import { render } from '@testing-library/react';
import { useNavigate, useParams } from 'react-router-dom';
import HeroList from './HeroList';

jest.mock('react-router-dom', () => ({
    useNavigate: jest.fn(),
    useParams: jest.fn(),
}));

describe('HeroList component', () => {
    beforeEach(() => {
        (useNavigate as jest.Mock).mockReturnValue(jest.fn());
        (useParams as jest.Mock).mockReturnValue({});
    });

    const mockHeroList = [
        { id: '1', name: 'Batman', image: 'https://via.placeholder.com/150' },
        { id: '2', name: 'Superman', image: 'https://via.placeholder.com/150' },
        {
            id: '3',
            name: 'Spiderman',
            image: 'https://via.placeholder.com/150',
        },
    ];

    it('should render a list of heroes', () => {
        const { getAllByTestId } = render(<HeroList heroList={mockHeroList} />);

        expect(getAllByTestId('hero-card-container')).toHaveLength(3);
        expect(getAllByTestId('hero-card-container')[0]).toHaveTextContent(
            'Batman'
        );
    });

    it('should render no heroes if the list is empty', () => {
        const { queryAllByTestId } = render(<HeroList heroList={[]} />);

        expect(queryAllByTestId('hero-card-container')).toHaveLength(0);
    });
});
