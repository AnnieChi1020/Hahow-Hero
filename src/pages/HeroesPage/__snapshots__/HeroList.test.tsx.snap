// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroList should render correctly 1`] = `
<div>
  <div
    class="sc-qRumB dehFKr"
  >
    <div
      class="sc-jTrPJq gXXERz"
      data-testid="hero-card-container"
    >
      <div
        class="sc-gLDzan jvOWUK"
      >
        <img
          class="sc-iAEyYk CCQck"
          src="https://via.placeholder.com/150"
        />
      </div>
      <div
        class="sc-fsQiph"
      >
        <h2
          class="sc-gueYoa dRSvwi"
        >
          Hero 1
        </h2>
      </div>
    </div>
    <div
      class="sc-jTrPJq gXXERz"
      data-testid="hero-card-container"
    >
      <div
        class="sc-gLDzan jvOWUK"
      >
        <img
          class="sc-iAEyYk CCQck"
          src="https://via.placeholder.com/150"
        />
      </div>
      <div
        class="sc-fsQiph"
      >
        <h2
          class="sc-gueYoa dRSvwi"
        >
          Hero 2
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeroList should render correctly when there are no heroes 1`] = `
<div>
  <div
    class="sc-qRumB dehFKr"
  />
</div>
`;
